Lorem Ipsum XR Translator/Assets/Scripts/Behaviors/SceneAnalyzer.cs

```diff
@@ -0,0 +1,226 @@
+
+        CameraParameters c = new CameraParameters();
+        c.hologramOpacity = 0.0f;
+        c.cameraResolutionWidth = cameraResolution.width;
+        c.cameraResolutionHeight = cameraResolution.height;
+        c.pixelFormat = CapturePixelFormat.BGRA32;
+
+        captureObject.StartPhotoModeAsync(c, OnPhotoModeStarted);
+
+    }
+    /// <summary>
+    /// Handle photo capture
+    /// </summary>
+    /// <param name="result">Whether the capture initialized successfully</param>
+    private void OnPhotoModeStarted(PhotoCapture.PhotoCaptureResult result)
+    {
+        if (result.success)
+        {
+            photoCaptureObject.TakePhotoAsync(OnCapturedPhotoToMemory);
+        }
+        else
+        {
+            DebugText.text = "Unable to start photo mode!";
+        }
+    }
+
+    /// <summary>
+    /// Function that handles what to do with captured photo data
+    /// </summary>
+    /// <param name="result">The result of the photo capture operation</param>
+    /// <param name="photoCaptureFrame">The captured photo data</param>
+    private void OnCapturedPhotoToMemory(PhotoCapture.PhotoCaptureResult result, PhotoCaptureFrame photoCaptureFrame)
+    {
+        if (result.success)
+        {
+
+            // Create our Texture2D for use and set the correct resolution
+            Resolution cameraResolution = PhotoCapture.SupportedResolutions.OrderByDescending((res) => res.width * res.height).First();
+            Texture2D targetTexture = new Texture2D(cameraResolution.width, cameraResolution.height);
+            // Copy the raw image data into our target texture
+            photoCaptureFrame.UploadImageDataToTexture(targetTexture);
+            this.image = targetTexture;
+            if (DebugQuad)
+            {
+                DebugQuad.GetComponent<Renderer>().material.mainTexture = targetTexture;
+            }
+            // Do as we wish with the texture such as apply it to a material, etc.
+            if (photoCaptureFrame.hasLocationData)
+            {
+                photoCaptureFrame.TryGetCameraToWorldMatrix(out Matrix4x4 cameraToWorldMatrix);
+
+                // This code was supposed to grab camera data out of PhotoCaptureFrame location data.
+                // It does not seem like this can be relied upon.
+                //this.worldMatrix = cameraToWorldMatrix;
+                //
+                //Vector3 position = cameraToWorldMatrix.GetColumn(3) - cameraToWorldMatrix.GetColumn(2);
+                //Quaternion rotation = Quaternion.LookRotation(-cameraToWorldMatrix.GetColumn(2), cameraToWorldMatrix.GetColumn(1));
+                //this.cameraPosition = position;
+                //
+                //photoCaptureFrame.TryGetProjectionMatrix(Camera.main.nearClipPlane, Camera.main.farClipPlane, out Matrix4x4 projectionMatrix);
+                //this.projectionMatrix = projectionMatrix;
```

Comment:
Do we still or are we going to need this?

Reply:
I think this can safely be removed. I think pulling the projection and matrix world from the main camera at the time of this function call will be sufficient. I'll take it out in the next commit.